pkg/provider/gcp/secretmanager/client.go

```diff
@@ -111,6 +113,7 @@ func (c *Client) DeleteSecret(ctx context.Context, remoteRef esv1beta1.PushRemot
 
 	deleteSecretVersionReq := &secretmanagerpb.DeleteSecretRequest{
 		Name: fmt.Sprintf("projects/%s/secrets/%s", c.store.ProjectID, remoteRef.GetRemoteKey()),
+		Etag: gcpSecret.Etag,
```

Comment:
Using [the Etag](https://cloud.google.com/secret-manager/docs/etags) would be better to prevent lost updates. Note that we cannot use the Etag for adding versions 😢